### PR TITLE
chef-nodes-chef.rb requires a client name set:

### PR DESCRIPTION
**Pull request by majormoses**
- made it required as an option
- added a sane default of `hostname --fqdn`

fixed CI issues with rubocop and dependencies for specific ruby versions

relates to: #13

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass 

#### Purpose

Basically the option was always required but it was not explicitly called out in option. Make a sane default of the hosts fqdn.

#### Known Compatablity Issues
This would be a breaking change other than the change in behavior will not really affect anyone negatively as you could not get it to work without using the value.

### CODE REVIEW

**Comment by majormoses** [comment]
the error is completely unrelated to the change: https://travis-ci.org/sensu-plugins/sensu-plugins-chef/jobs/197506306#L170-L185

**Comment by majormoses** [comment]
Modified travis build in order to get builds passing (hopefully).

**Comment by majormoses** [comment]
existing dependencies were broken for ruby versions < 2.2 working on fixing it up

**Comment by majormoses** [comment]
@eheydrick can we get this merged and released? I think this should fix most of the issues people are seeing in #13 I have not looked into the comment regarding excluded hosts but I think that should be addressed in a separate PR.

**Comment by eheydrick** [comment]
I like the client name default. Let's remove Ruby 2.0 support. It would simplify the dependencies somewhat and we've only been keeping it around because the windows client shipped with it. But in 0.27 windows ships with 2.3 so we should start dropping 2.0 support. Can you also combine the dependencies for the same ruby version together so it's easier to read instead of separate conditionals.

**Comment by majormoses** [comment]
@eheydrick dropped ruby 2.0 and consolidated to 2 conditionals. Let me know if there is anything else.
